### PR TITLE
Include minikube setup instructions

### LINKED ISSUE #332: [minikube] Provide support for using OpenEBS with mini-kube.
The current vagrant deployment launches 3 VMs for trying out OpenEBS. This will require that a developer has a laptop which has 8GB RAM and 8 cores at the least. And with these 3 VMs running, there may not be any resources left to run/compile the code on the laptop. 

For developer consumption, running ut or experimenting, there is a need to bring up openebs on mini-kube.

### LINKED ISSUE #350: Add Developer Solutions section under Setup
Add Developer Solutions heading and have 
Setup OpenEBS Using Minikube and Performance Benchmarking sections

### LINKED ISSUE #351: Add content for Setup OpenEBS using Minikube
Add content.

### PR DESCRIPTION
**Pull request by kmova**
Fixes #332 - validates that openebs can be run with minikube
Fixes #350 - developer solutions category under setup
Fixes #351 - sample instructions for setting up minikube

Adds new document (rst) file for documentation developer solutions
like minikube. This section will need to be further improved with
links to developer options like modifying the PVC yaml files
with openebs annotations.

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: documentation/source/install/dev_solutions.rst [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+Developer Solutions
+*******************
+
+Minikube
+========
+
+Setting up OpenEBS with Kubernetes using Minikube
+-------------------------------------------------
+
+Minikube helps developers to quickly setup a single-node Kubernetes cluster for their development environement. There are several options available for developers to install Minikube. See `Minikube Setup`_.
```
Comment (mahebbar) [inline documentation/source/install/dev_solutions.rst]: environment spelling

Comment (mahebbar) [reply documentation/source/install/dev_solutions.rst]: See, (add comma)

File: documentation/source/install/dev_solutions.rst [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+       .. _Minikube Setup: https://github.com/kubernetes/minikube
+
+If you are already an experienced Minikube user, you can easily setup OpenEBS on your existing Kubernetes cluster with a few simple kubectl commands. See, :ref:`quick-start`.
+
+This section provides instructions to set up Kubernetes using Minikube directly on Ubuntu 16.06 (without using any VM drivers) and to have OpenEBS running in hyper converged mode.
```
Comment (mahebbar) [inline documentation/source/install/dev_solutions.rst]: hyperconverged

File: documentation/source/install/dev_solutions.rst [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+If you are already an experienced Minikube user, you can easily setup OpenEBS on your existing Kubernetes cluster with a few simple kubectl commands. See, :ref:`quick-start`.
+
+This section provides instructions to set up Kubernetes using Minikube directly on Ubuntu 16.06 (without using any VM drivers) and to have OpenEBS running in hyper converged mode. 
+
+Prerequisites
+-------------
+Minimum requirements for using Minikube
+
+* Machine Type - (Minimum 4 vCPUs)
+* RAM - (Minimum 4GB)
```
Comment (mahebbar) [inline documentation/source/install/dev_solutions.rst]: 4 GB

File: documentation/source/install/dev_solutions.rst [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+Make sure *docker* is installed on your Ubuntu Host.
```
Comment (mahebbar) [inline documentation/source/install/dev_solutions.rst]: host

File: documentation/source/install/dev_solutions.rst [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+Check that initiator name is configured and iSCSI service is running using the following commands.
+::
+   sudo cat /etc/iscsi/initiatorname.iscsi
+   sudo service open-iscsi status
+
+
+Download and setup Minikube and kubectl
+---------------------------------------
+
+On your Ubuntu host, install minikube
```
Comment (mahebbar) [inline documentation/source/install/dev_solutions.rst]: full stop at the end of the sentence.

File: documentation/source/install/dev_solutions.rst [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+::
+    curl -Lo minikube https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64
+    chmod +x minikube 
+    sudo mv minikube /usr/local/bin/
+
+On your Ubuntu host, install kubectl
```
Comment (mahebbar) [inline documentation/source/install/dev_solutions.rst]: full stop at the end of the sentence.

File: documentation/source/install/dev_solutions.rst [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+::
+    curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl
+    chmod +x kubectl 
+    sudo mv kubectl /usr/local/bin/
+
+On your Ubuntu host, setup directories for storing minkube and kubectl configuration
```
Comment (mahebbar) [inline documentation/source/install/dev_solutions.rst]: full stop at the end of the sentence.

File: documentation/source/install/dev_solutions.rst [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+::
+    mkdir /home/vagrant/.kube || true
+    touch /home/vagrant/.kube/config
+
+On your Ubuntu host, setup ENV for minikube. Copy the following to ~/.profile
```
Comment (mahebbar) [inline documentation/source/install/dev_solutions.rst]: full stop at the end of the sentence.

File: documentation/source/install/dev_solutions.rst [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+::
+    export MINIKUBE_WANTUPDATENOTIFICATION=false
+    export MINIKUBE_WANTREPORTERRORPROMPT=false
+    export MINIKUBE_HOME=$HOME
+    export CHANGE_MINIKUBE_NONE_USER=true
+    export KUBECONFIG=$HOME/.kube/config
+
+On your Ubuntu host, start the minikube
```
Comment (mahebbar) [inline documentation/source/install/dev_solutions.rst]: full stop at the end of the sentence.

File: documentation/source/install/dev_solutions.rst [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+    sudo -E minikube start --vm-driver=none
+
+Verify that minikube is configured
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Check that minikube is configured and it has started using the following commands.
+::
+    minikube status
+
+When minikube is configured properly, *minikube status* will provide output as follows:
```
Comment (mahebbar) [inline documentation/source/install/dev_solutions.rst]: remove properly? 
will display the following output?

File: documentation/source/install/dev_solutions.rst [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+::
+   minikube: Running
+   cluster: Running
+   kubectl: Correctly Configured: pointing to minikube-vm at 127.0.0.1
+
+**Note** 
+In case, the minikube errors our with permission denied message to configuration files, run the following commands.
```
Comment (mahebbar) [inline documentation/source/install/dev_solutions.rst]: If Minikube displays errors with permission....

File: documentation/source/install/dev_solutions.rst [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+::
+    sudo chown -R $USER $HOME/.kube
+    sudo chgrp -R $USER $HOME/.kube
+    sudo chown -R $USER $HOME/.minikube
+    sudo chgrp -R $USER $HOME/.minikube
+
+Verify that kubernetes is configured
```
Comment (mahebbar) [inline documentation/source/install/dev_solutions.rst]: Kubernetes

File: documentation/source/install/dev_solutions.rst [hunk 13]
```diff
@@ -0,0 +1,156 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Check that kubectl is configured and services are running using the following commands.
+::
+    kubectl get pods
+    kubectl get nodes
+
+When configured properly, the above kubectl command will provide output similar to following:
```
Comment (mahebbar) [inline documentation/source/install/dev_solutions.rst]: remove 'properly'
will display output similar....

File: documentation/source/install/dev_solutions.rst [hunk 14]
```diff
@@ -0,0 +1,156 @@
...
+
+Get the list of storage classes using the following command. Choose the storage class that best suits your application.
+::
+   kubectl get sc
+
+Some sample YAML files for stateful workloads using OpenEBS are provided in the `openebs/k8s/demo`_
+        
+  .. _openebs/k8s/demo: https://github.com/openebs/openebs/tree/master/k8s/demo
+
+By default, OpenEBS launches OpenEBS Volumes with two replicas. To set one replica, as is the case with single-node Kubernetes cluster, modify the PVC spec of your applications as follows:
```
Comment (mahebbar) [inline documentation/source/install/dev_solutions.rst]: spec = specification

**Comment by kmova** [comment]
Thanks @AmitKumarDas  for pointing out that PVC spec updated in the previous commit (339c2f7)
 is for maya-apiserver and is not the same as the PVC used by the kubernetes. Fixed this as per your suggestion to use ENV (f9c7196).

File: documentation/source/install/dev_solutions.rst (hunk 1 above)
Comment (kmova) [reply documentation/source/install/dev_solutions.rst]: thanks!

File: documentation/source/install/dev_solutions.rst (hunk 2 above)
Comment (kmova) [reply documentation/source/install/dev_solutions.rst]: thanks!

File: documentation/source/install/dev_solutions.rst (hunk 3 above)
Comment (kmova) [reply documentation/source/install/dev_solutions.rst]: thanks!

File: documentation/source/install/dev_solutions.rst (hunk 4 above)
Comment (kmova) [reply documentation/source/install/dev_solutions.rst]: done!

File: documentation/source/install/dev_solutions.rst (hunk 9 above)
Comment (kmova) [reply documentation/source/install/dev_solutions.rst]: added periods.. for all the above mentioned lines

File: documentation/source/install/dev_solutions.rst (hunk 10 above)
Comment (kmova) [reply documentation/source/install/dev_solutions.rst]: there chances.. of minikube configuration errors. so, i will leave properly for now. made the other sentence change.

File: documentation/source/install/dev_solutions.rst (hunk 11 above)
Comment (kmova) [reply documentation/source/install/dev_solutions.rst]: good one! changed

File: documentation/source/install/dev_solutions.rst (hunk 12 above)
Comment (kmova) [reply documentation/source/install/dev_solutions.rst]: yes.. done

File: documentation/source/install/dev_solutions.rst (hunk 13 above)
Comment (kmova) [reply documentation/source/install/dev_solutions.rst]: done

**Pull request merged**